mlflow/utils/_spark_utils.py

```diff
@@ -45,9 +72,10 @@ def _create_local_spark_session_for_recipes():
         # Return None if user doesn't have PySpark installed
         return None
     _prepare_subprocess_environ_for_creating_local_spark_session()
+    spark_scala_version = _get_spark_scala_version()
```

Comment:
Can we have try-catch to show reasonable error message here or perhaps fallback to hard-coded version?

Reply:
I refined exception , but I don't make it fallback to hard-coded value, because hard-coded value might be wrong and mismatched scala version would cause more issues which are hard to debug.